lib/src/main/java/com/diffplug/spotless/npm/Reflective.java

```diff
@@ -0,0 +1,279 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.npm;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.reflect.*;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+class Reflective {
+	private final ClassLoader classLoader;
+
+	private Reflective(ClassLoader classLoader) {
+		this.classLoader = requireNonNull(classLoader);
+	}
+
+	static Reflective withClassLoader(ClassLoader classLoader) {
+		return new Reflective(classLoader);
+	}
+
+	Class<?> clazz(String className) {
+		try {
+			return this.classLoader.loadClass(className);
+		} catch (ClassNotFoundException e) {
+			throw new ReflectiveException(e);
+		}
+	}
+
+	private Method staticMethod(String className, String methodName, Object... parameters) {
+		try {
+			final Class<?> clazz = clazz(className);
+			return clazz.getDeclaredMethod(methodName, types(parameters));
+		} catch (NoSuchMethodException e) {
+			throw new ReflectiveException(e);
+		}
+	}
+
+	Object invokeStaticMethod(String className, String methodName, Object... parameters) {
+		try {
+			Method m = staticMethod(className, methodName, parameters);
+			return m.invoke(m.getDeclaringClass(), parameters);
+		} catch (IllegalAccessException | InvocationTargetException e) {
+			throw new ReflectiveException(e);
+		}
+	}
+
+	private Class<?>[] types(TypedValue[] typedValues) {
+		return Arrays.stream(typedValues)
+				.map(TypedValue::getClazz)
+				.toArray(Class[]::new);
+	}
+
+	Class<?>[] types(Object[] arguments) {
+		return Arrays.stream(arguments)
+				.map(Object::getClass)
+				.toArray(Class[]::new);
+	}
+
+	Object invokeMethod(Object target, String methodName, Object... parameters) {
+		Method m = method(target, clazz(target), methodName, parameters);
+		try {
+			return m.invoke(target, parameters);
+		} catch (IllegalAccessException | InvocationTargetException e) {
+			throw new ReflectiveException(e);
+		}
+	}
+
+	Object invokeMethod(Object target, String methodName, TypedValue... parameters) {
+		Method m = method(target, clazz(target), methodName, parameters);
+		try {
+			return m.invoke(target, objects(parameters));
+		} catch (IllegalAccessException | InvocationTargetException e) {
+			throw new ReflectiveException(e);
+		}
+	}
+
+	private Method method(Object target, Class<?> clazz, String methodName, Object[] parameters) {
+		try {
+			final Method method = findMatchingMethod(clazz, methodName, parameters);
+			return method;
+		} catch (NoSuchMethodException e) {
+			if (clazz.getSuperclass() != null) {
+				return method(target, clazz.getSuperclass(), methodName, parameters);
+			} else {
+				// try with primitives
```

Comment:
Remove?

Reply:
you mean the empty line, correct?

Reply:
I meant the comment, here is not trying primitives, it's throwing an exception.